modules/tensor_mechanics/src/materials/RecomputeRadialReturnStressIncrement.C

```diff
@@ -0,0 +1,117 @@
+    _absolute_tolerance(parameters.get<Real>("absolute_tolerance"))
+{
+}
+
+void
+RecomputeRadialReturnStressIncrement::computeStressIncrement()
+{
+  // Given the stretching, compute the stress increment and add iteration to the old stress. Also update the inelastic strain
+  // Compute the stress in the intermediate configuration while retaining the stress history
+  _return_stress_increment[_qp] = _elasticity_tensor[_qp] * _strain_increment[_qp] + _stress_old[_qp];
+
+  // compute the deviatoric trial stress and trial strain from the current intermediate configuration
+  RankTwoTensor deviatoric_trial_stress = _return_stress_increment[_qp].deviatoric();
+
+  // compute the effective trial stress
+  Real dev_trial_stress_squared = deviatoric_trial_stress.doubleContraction(deviatoric_trial_stress);
+  Real effective_trial_stress = std::sqrt(3.0 / 2.0 * dev_trial_stress_squared);
+
+  computeStressInitialize(effective_trial_stress);
+
+  // Use Newton sub-iteration to determine the scalar effective inelastic strain increment
+  Real scalar_effective_inelastic_strain = 0;
+  unsigned int iteration = 0;
+  Real residual = 10;
+  Real norm_residual = 10;
+  Real first_norm_residual = 10;
+
+  // create an output string with iteration information when errors occur
+  std::string iteration_output;
+
+  while (iteration < _max_its &&
+        norm_residual > _absolute_tolerance &&
+        (norm_residual/first_norm_residual) > _relative_tolerance)
+  {
+    iterationInitialize(scalar_effective_inelastic_strain);
+
+    residual = computeResidual(effective_trial_stress, scalar_effective_inelastic_strain);
+    norm_residual = std::abs(residual);
+    if (iteration == 0)
+    {
+      first_norm_residual = norm_residual;
+      if (first_norm_residual == 0) // shouldn't be needed here; already set to 10
+        first_norm_residual = 1;
+    }
+
+    Real derivative = computeDerivative(effective_trial_stress, scalar_effective_inelastic_strain);
+
+    scalar_effective_inelastic_strain -= residual / derivative;
+
+    if (_output_iteration_info || _output_iteration_info_on_error)
+    {
+      iteration_output = "In the element " + Moose::stringify(_current_elem->id()) +
+                       + " and the qp point " + Moose::stringify(_qp) + ": \n" +
+                       + " iteration = " + Moose::stringify(iteration ) + "\n" +
+                       + " effective trial stress = " + Moose::stringify(effective_trial_stress) + "\n" +
+                       + " scalar effective inelastic strain = " + Moose::stringify(scalar_effective_inelastic_strain) +"\n" +
+                       + " relative residual = " + Moose::stringify(norm_residual/first_norm_residual) + "\n" +
+                       + " relative tolerance = " + Moose::stringify(_relative_tolerance) + "\n" +
+                       + " absolute residual = " + Moose::stringify(norm_residual) + "\n" +
+                       + " absolute tolerance = " + Moose::stringify(_absolute_tolerance);
```

Comment:
there is no `\n` at the end of the added iteration output, so if multiple outputs are concatenated they are mangled together.